internal/daemon/controller/handlers/authmethods/grants_test.go

```diff
@@ -635,575 +634,721 @@ func TestGrants_WriteActions(t *testing.T) {
 				},
-				canUpdateAuthMethod: map[string]expectedOutput{
+			}),
+			canUpdateAuthMethod: func(t *testing.T) map[string]expectedOutput {
+				globalAmId := password.TestAuthMethod(t, conn, globals.GlobalPrefix).PublicId
+				org1AmId := password.TestAuthMethod(t, conn, org1.PublicId).PublicId
+				org2AmId := password.TestAuthMethod(t, conn, org2.PublicId).PublicId
+				return map[string]expectedOutput{
 					globalAmId: {wantOutfields: []string{globals.IdField, globals.NameField, globals.DescriptionField, globals.VersionField}},
 					org1AmId:   {wantOutfields: []string{globals.IdField, globals.ScopeIdField, globals.TypeField, globals.VersionField}},
 					org2AmId:   {wantErr: handlers.ForbiddenError()},
-				},
+				}
 			},
-			{
-				name: "org role can't update global auth methods",
-				rolesToCreate: []authtoken.TestRoleGrantsForToken{{
-					RoleScopeId:  org1.PublicId,
-					GrantStrings: []string{"ids=*;type=auth-method;actions=update;output_fields=id,version,created_time,updated_time"},
-					GrantScopes:  []string{globals.GrantScopeThis},
-				}},
-				canUpdateAuthMethod: map[string]expectedOutput{
+		},
+		{
+			name: "org role can't update global auth methods",
+			userFunc: iam.TestUserDirectGrantsFunc(t, conn, kmsCache, globals.GlobalPrefix, password.TestAuthMethodWithAccount, []iam.TestRoleGrantsRequest{
+				{
+					RoleScopeId: org1.PublicId,
+					Grants:      []string{"ids=*;type=auth-method;actions=update;output_fields=id,version,created_time,updated_time"},
+					GrantScopes: []string{globals.GrantScopeThis},
+				},
+			}),
+			canUpdateAuthMethod: func(t *testing.T) map[string]expectedOutput {
+				globalAmId := password.TestAuthMethod(t, conn, globals.GlobalPrefix).PublicId
+				org1AmId := password.TestAuthMethod(t, conn, org1.PublicId).PublicId
+				org2AmId := password.TestAuthMethod(t, conn, org2.PublicId).PublicId
+				return map[string]expectedOutput{
 					globalAmId: {wantErr: handlers.ForbiddenError()},
 					org1AmId:   {wantOutfields: []string{globals.IdField, globals.VersionField, globals.CreatedTimeField, globals.UpdatedTimeField}},
 					org2AmId:   {wantErr: handlers.ForbiddenError()},
-				},
+				}
 			},
-			{
-				name: "global role grant children of global auth method's id can only update children auth methods",
-				rolesToCreate: []authtoken.TestRoleGrantsForToken{
-					{
-						RoleScopeId:  globals.GlobalPrefix,
-						GrantStrings: []string{"ids=" + globalAmId + ";type=auth-method;actions=update;output_fields=id"},
-						GrantScopes:  []string{globals.GrantScopeChildren},
-					},
+		},
+		{
+			name: "global role grant children of global auth method's id can only update children auth methods",
+			userFunc: iam.TestUserDirectGrantsFunc(t, conn, kmsCache, globals.GlobalPrefix, password.TestAuthMethodWithAccount, []iam.TestRoleGrantsRequest{
+				{
+					RoleScopeId: globals.GlobalPrefix,
+					Grants:      []string{"ids=*;type=auth-method;actions=update;output_fields=id"},
```

Comment:
This testcase intended to test a grant string with a pinned id, but we can't do that anymore with the current setup 🤔